tree/dataframe/inc/ROOT/RDF/RMergeableValue.hxx

```diff
@@ -0,0 +1,76 @@
+#ifndef ROOT_RDF_RMERGEABLEVALUE
+#define ROOT_RDF_RMERGEABLEVALUE
+
+#include <functional>
+#include <memory>
+#include "RtypesCore.h"
+
+namespace ROOT {
+namespace RDF {
+
+template <typename T>
+class RMergeableValue {
+protected:
+   T fValue;
+
+public:
+   RMergeableValue() = delete;
+   RMergeableValue(T &value) : fValue{value} {}
+   virtual ~RMergeableValue() = default;
+   virtual void Merge(const std::unique_ptr<RMergeableValue<T>> &) = 0;
+   virtual ULong64_t GetCounts() { return 0; }
```

Comment:
Because in `RMergeableMean::Merge` :
```
   void Merge(const std::unique_ptr<RMergeableValue<T>> &other) override
   {
      auto &othervalue = other->GetValue();
      const auto &othercounts = other->GetCounts();
```
So with this signature `RMergeableValue` needs to have a `GetCounts` method

Reply:
you can cast `other` to `RMeanMergeableValue<T>` or whatever type it must be and only give `GetCounts()` to the sub-type that needs it. (in fact you don't even need a getter because instances of the same type see each other's private data members)

Reply:
The cast does the trick in fact. I still prefer to leave the getter with a `const` signature